pytorch_lightning/trainer/connectors/checkpoint_connector.py

```diff
@@ -400,11 +400,11 @@ def save_checkpoint(self, filepath, weights_only: bool = False):
         """
         # dump states as a checkpoint dictionary object
         checkpoint = self.dump_checkpoint(weights_only)
-
         if self.trainer.is_global_zero:
             # write the checkpoint dictionary on the file
-            if self.trainer.accelerator_backend:
-                checkpoint = self.trainer.accelerator_backend.on_save(checkpoint)
+
+            if self.trainer.training_type_plugin:
```

Comment:
trainer/accelerator always has a traintype plugin, so we could even drop the if condition :)

Reply:
I think during sanity check the `accelerator_backend` was not set. Not sure if that's the case now.